keps/sig-auth/3766-referencegrant/README.md

```diff
@@ -441,174 +368,358 @@ rejected".
 // * A reference to Namespace and Resource that exists but a ReferenceGrant
 //   allowing the reference does not exist
 //
-// If any of the above occur, a generic error message such as "RefNotPermitted" 
+// If any of the above occur, a generic error message such as "RefNotPermitted"
 // should be communicated, likely via status on the referring resource.
 //
 // Support: Core
 type ReferenceGrant struct {
-	metav1.TypeMeta   `json:",inline"`
-	metav1.ObjectMeta `json:"metadata,omitempty"`
-
-	// Spec defines the desired state of ReferenceGrant.
-	Spec ReferenceGrantSpec `json:"spec,omitempty"`
-
-	// Note that `Status` sub-resource has been excluded at the
-	// moment as it was difficult to work out the design.
-	// A `Status` sub-resource may be added in the future.
+  metav1.TypeMeta   `json:",inline"`
+  metav1.ObjectMeta `json:"metadata,omitempty"`
+
+  // From describes the trusted namespaces and resources that can reference the
+  // resources described in "To". Each entry in this list MUST be considered
+  // to be an additional place that references can be valid from, or to put
+  // this another way, entries MUST be combined using OR.
+  //
+  From ReferenceGrantFrom `json:"from"`
+
+  // To describes the resources in this namespace that may be referenced by
+  // the resources described in "From". Each entry in this list MUST be
+  // considered to be an additional set of objects that references can be
+  // valid to, or to put this another way, entries MUST be combined using OR.
+  //
+  // +kubebuilder:validation:MinItems=1
+  // +kubebuilder:validation:MaxItems=16
+  To []ReferenceGrantTo `json:"to"`
+
+  // Verbs describes the actions allowed by references using this grant.
+  // The following verbs are valid:
+  //
+  // * "ForwardTraffic": Allow forwarding traffic from resources described in
+  //                     "from" to resources described in "to".
+  // * "UseTLSCert":     Allow TLS Certificates found in resources described in
+  //                     "to" to be used for TLS termination by resources described
+  //                     in "from".
+  // * "Backup":         Allow data contained in resources described in "to" to be
```

Comment:
Is this supposed to be for the cross-namespace snapshot restore case? If so, then I think a better verb may be PopulateData:  Resources described in "to" can be referenced as a data source for resources described in "from"

Reply:
Good call, updated, thanks!